lib/src/cpp/utils/XMLManager.cpp

```diff
@@ -177,9 +174,13 @@ void XMLManager::transform_standalone_to_rooted_structure()
 
 void XMLManager::validate_and_save_document()
 {
+    // Set the write required as false by default
+    write_required = false;
+
+    // Set write as required if XML is valid
```

Comment:
I suppose we should start thinking about the `print` method which would not modify the file. But it can be done when we first introduce a print function.

Reply:
I think it will not be an issue.
The method would look like this:
```print_node()
{
    return xercesc::XMLString::transcode(serializer->writeToString(last_node));
}```